libopenage/renderer/opengl/geometry.cpp

```diff
@@ -51,6 +51,10 @@ void GlGeometry::update_verts_offset(std::vector<uint8_t> const &verts, size_t o
 	this->mesh->vertices.upload_data(verts.data(), offset, verts.size());
 }
 
+GLuint GlGeometry::get_vao_handle() const {
+	return this->mesh->vao.get_handle();
```

Comment:
This should check the geometry type and throw an informative exception if it's not `geometry_t::mesh`.